library/marathon.py

```diff
@@ -187,12 +193,32 @@ def update_app(self, json_definition):
             self._marathon_client.update_app(self._appid, app)
             return app_json, True
 
+    def diff_app(self, json_definition, compact_diff=False):
+        deployed_app = self._get_app_info()
+        if deployed_app:
+            deployed_app_json_obj = json.loads(deployed_app.to_json())
+        else:
+            deployed_app_json_obj = json.loads('{}')
+        json_definition_obj = json.loads(json_definition)
+        string_old = json.dumps(deployed_app_json_obj, sort_keys=True, indent=4, separators=('', ': '))
+        string_new = json.dumps(json_definition_obj, sort_keys=True, indent=4, separators=('', ': '))
+        diff = ('\n '.join((difflib.unified_diff(string_old.split('\n'), string_new.split('\n'))))).split('\n')
+        if compact_diff:
+            # Collecting just the lines indicating a difference
+            result = []
+            for line in diff:
+                if line.strip().startswith('-') or line.strip().startswith('+'):
+                    result.append(line)
+            diff = result
```

Comment:
(opt) above block would be nice candidate for a functional  approach (with the filter func)